pkg/rancher-desktop/config/__tests__/commandLineOptions.spec.ts

```diff
@@ -0,0 +1,377 @@
+      const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--kubernetes.options.flannel']);
+
+      expect(origPrefs.kubernetes.options.flannel).toBeFalsy();
+      expect(newPrefs.kubernetes.options.flannel).toBeTruthy();
+      newPrefs.kubernetes.options.flannel = false;
+      expect(newPrefs).toEqual(origPrefs);
+    });
+
+    test('boolean option set to false should change only that value', () => {
+      const newPrefs = updateFromCommandLine(prefs, lockedSettings, ['--kubernetes.options.traefik=false']);
+
+      expect(origPrefs.kubernetes.options.traefik).toBeTruthy();
+      expect(newPrefs.kubernetes.options.traefik).toBeFalsy();
+      newPrefs.kubernetes.options.traefik = true;
+      expect(newPrefs).toEqual(origPrefs);
+    });
+
+    test('changes specified options', () => {
+      const optionsByPlatform: Record<string, Array<string | [string, string]>> = {
+        win32: [
+          '--virtualMachine.hostResolver',
+        ],
+        '!win32': [
+          '--application.adminAccess',
+          ['--application.pathManagementStrategy', 'rcfiles'],
+          '--virtualMachine.memoryInGB',
+          '--virtualMachine.numberCPUs',
+        ],
+        darwin: [
+          '--experimental.virtualMachine.socketVMNet',
+        ],
+        '*': [
+          '--application.debug',
+          '--application.telemetry.enabled',
+          '--application.updater.enabled',
+          '--application.autoStart',
+          '--application.startInBackground',
+          '--application.hideNotificationIcon',
+          '--application.window.quitOnClose',
+          '--containerEngine.allowedImages.enabled',
+          ['--containerEngine.name', 'containerd'],
+          '--kubernetes.port',
+          '--kubernetes.enabled',
+          '--kubernetes.options.traefik',
+          '--kubernetes.options.flannel',
+          '--portForwarding.includeKubernetesServices',
+          '--images.showAll',
+          ['--images.namespace', 'mangos'],
+          '--diagnostics.showMuted',
+        ],
+      };
+
+      for (const platform in optionsByPlatform) {
+        const options: Array<string | [string, string]> = optionsByPlatform[platform as 'win32'|'linux'|'darwin'|'*'];
+
+        for (const entry of options) {
+          let option: string;
+          let newValue: string|boolean|number|undefined;
+
+          if (Array.isArray(entry)) {
+            option = entry[0];
+            newValue = entry[1];
+          } else {
+            option = entry;
+          }
+          const accessor = option.substring(2);
+          const oldValue: string|boolean|number|undefined = _.get(origPrefs, accessor);
+
+          if (oldValue === undefined) {
+            console.log('stop here');
+          }
+          expect(oldValue).not.toBeUndefined();
+          if (newValue === undefined) {
+            expect(['boolean', 'number']).toContain(typeof oldValue);
```

Comment:
Consider moving this into the `switch` as a `default:`, so that the switch covers all cases.

Reply:
FWICT it's marginally more efficient to just make the assertion in the default stmt, but ok.

Reply:
This is because I'm hoping to turn on the [`switch-exhaustiveness-check`](https://typescript-eslint.io/rules/switch-exhaustiveness-check) linter, so doing this now would make that transition easier :)